### PR TITLE
2185 cannot create expression protein

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

File: src/PKSim.Presentation/DTO/ExpressionProfiles/ExpressionProfileDTO.cs [hunk 1]
```diff
@@ -60,35 +58,23 @@ private static class ExpressionProfileRules
 
          private static IBusinessRule speciesNotNull { get; } = GenericRules.NotNull<ExpressionProfileDTO, Species>(x => x.Species, PKSimConstants.Error.SpeciesIsRequired);
 
-         private static IBusinessRule moleculeNameValid { get; } = CreateRule.For<ExpressionProfileDTO>()
```
Comment (msevestre) [inline src/PKSim.Presentation/DTO/ExpressionProfiles/ExpressionProfileDTO.cs:63]: @Yuri05 Removing those rules that are misleading

File: src/PKSim.Presentation/DTO/ExpressionProfiles/ExpressionProfileDTO.cs [hunk 2]
```diff
@@ -60,35 +58,23 @@ private static class ExpressionProfileRules
...
-         private static IBusinessRule categoryValid { get; } = CreateRule.For<ExpressionProfileDTO>()
-            .Property(item => item.Category)
-            .WithRule((dto, category) => dto.moleculeSpeciesCategoryValid(dto.MoleculeName, dto.Species, category))
-            .WithError((dto, category) => PKSimConstants.Error.NameAlreadyExistsInContainerType(category, PKSimConstants.ObjectTypes.Project));
-
-         private static IBusinessRule speciesValid { get; } = CreateRule.For<ExpressionProfileDTO>()
-            .Property(item => item.Species)
-            .WithRule((dto, species) => dto.moleculeSpeciesCategoryValid(dto.MoleculeName, species, dto.Category))
-            .WithError((dto, species) => PKSimConstants.Error.NameAlreadyExistsInContainerType(species.Name, PKSimConstants.ObjectTypes.Project));
+         private static IBusinessRule nameValid { get; } = CreateRule.For<ExpressionProfileDTO>()
```
Comment (msevestre) [inline src/PKSim.Presentation/DTO/ExpressionProfiles/ExpressionProfileDTO.cs:61]: instead only one with the name

File: src/PKSim.Presentation/DTO/ExpressionProfiles/ExpressionProfileDTO.cs [hunk 3]
```diff
@@ -60,35 +58,23 @@ private static class ExpressionProfileRules
...
-            yield return categoryValid;
-            yield return speciesValid;
+            yield return nameValid;
          }
       }
 
       public void AddExistingExpressionProfileNames(IEnumerable<string> existingNames)
       {
-         _allExistingExpressionProfileNames.AddRange(existingNames);
+         _allExistingExpressionProfileNames.AddRange(existingNames.Select(x => x.ToLower()));
```
Comment (msevestre) [inline src/PKSim.Presentation/DTO/ExpressionProfiles/ExpressionProfileDTO.cs:77]: makign sure lowercase combination does not work also

File: src/PKSim.UI/Views/ExpressionProfiles/CreateExpressionProfileView.cs [hunk 4]
```diff
@@ -76,11 +78,19 @@ public override void InitializeResources()
          layoutItemSpecies.Text = PKSimConstants.UI.Species.FormatForLabel();
          labelCategoryDescription.AsDescription();
          labelCategoryDescription.Text = PKSimConstants.UI.ExpressionProfileCategoryDescription.FormatForDescription();
+
+         //Do not close on OK
+         ButtonOk.DialogResult = DialogResult.None;
```
Comment (msevestre) [inline src/PKSim.UI/Views/ExpressionProfiles/CreateExpressionProfileView.cs:82]: This is the key here. Overriding the behavior of OK to do our own stuff

**Comment by Yuri05** [comment]
Looks good for me. Just one question **rename** protein expression is also handled?

**Comment by msevestre** [comment]
no it's not. Update coming

**Comment by msevestre** [comment]
@Yuri05 Merging

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**